master/src/main/java/org/apache/celeborn/service/deploy/master/MasterSecretRegistryImpl.java

```diff
@@ -44,7 +44,32 @@ public void register(String appId, String secret) {
     }
   }
 
-  void setMetadataHandler(IMetadataHandler metadataHandler) {
+  @Override
+  public String getSecretKey(String appId) {
+    String secret = super.getSecretKey(appId);
+    if (secret == null && metadataHandler != null) {
+      LOG.info("Fetching secret from AbstractMetaManager for appId: {}", appId);
```

Comment:
oh, probably just say metadata manager? (not to expose a class name)

Reply:
thanks! changed to metadata manager